lib/matplotlib/cbook/__init__.py

```diff
@@ -1571,6 +1608,219 @@ def pts_to_prestep(x, *args):
+                diff_root_x = np.interp(
+                    0, diff_values[diff_order], x_values[diff_order])
+                x_order = x_values.argsort()
+                diff_root_y = np.interp(diff_root_x, x_values[x_order],
+                                        y1_values[x_order])
+                return diff_root_x, diff_root_y
+
+            start = get_interp_point(ind0)
+            end = get_interp_point(ind1)
+        else:
+            # the purpose of the next two lines is for when y2 is a
+            # scalar like 0 and we want the fill to go all the way
+            # down to 0 even if none of the y1 sample points do
+            start = xslice[0], y2slice[0]
+            end = xslice[-1], y2slice[-1]
+
+        X[0] = start
+        X[N + 1] = end
+
+        X[1:N + 1, 0] = xslice
+        X[1:N + 1, 1] = y1slice
+        X[N + 2:, 0] = xslice[::-1]
+        X[N + 2:, 1] = y2slice[::-1]
+
+        if dir == 'y':
+            polys.append(X)
+        elif dir == 'x':
+            polys.append(X.T[::-1].T)
+    return polys
+
+
+def pts_to_betweenstep(x, *args):
+    """
+    Convert continuous line to between-steps.
+
+    Given a set of ``N`` edges and ``N - 1`` values padded with Nan to size N,
+    converts to ``2N - 2`` points, which when connected linearly give
+    a step function connecting step edges at a specified value
+
+    Parameters
+    ----------
+    x : array
+        The x location of the step edges. May be empty.
+
+    y1, ..., yp : array
+        y arrays to be turned into steps; must have length as ``x`` +/- 1.
+    Returns
+    -------
+    out : array
+        The x and y values converted to steps in the same order as the input;
+        can be unpacked as ``x_out, y1_out, ..., yp_out``.  If the input is
+        length ``N``, each of these arrays will be length ``2N - 2``. For
+        ``N=0``, the length will be 0.
+
+    """
+    args = np.array(args)
+    step_length = max(2 * max(len(x), len(args[0])) - 2, 0)
+    steps = np.zeros((1 + len(args), step_length))
+
+    xlike = len(x) == len(args[0]) + 1
+    ylike = len(x) + 1 == len(args[0])
+
+    def __between__(steps, sl0, sl1, _x, _args):
+        # Be agnostic whether xlike or ylike
+        if _x.flatten().shape != x.shape:
+            if _x.flatten().shape[-1] == _x.shape[-1]:
+                _x = _x[0]
+        steps[sl0, ::steps.shape[-1]-1] = _x[::_x.shape[-1]-1]
+        steps[sl0, 2::2] = _x[1:-1]
+        steps[sl0, 1:-1:2] = _x[1:-1]
+        steps[sl1, 0::2] = _args
+        steps[sl1, 1::2] = _args
+        return steps
+
+    if xlike:
+        steps = __between__(steps, 0, slice(1, None),
+                            x, args)
+    elif ylike:
+        steps = __between__(steps, slice(1, None), 0,
+                                args, x)
+    else:
+        # Fall back to steps-post (e.g. legend drawing)
+        steps = pts_to_poststep(x, *args)
+
+    return steps
+
+
+def pts_to_betweenstep_edges(x, *args):
```

Comment:
This should be private...

Reply:
Like I acknowledge the other ones aren't private, but that was a mistake, so we shouldn't keep making it worse.

Reply:
Hmm, alright, but shouldn't it all be fixed at once? There are enough changes as is in this PR.

Reply:
No you should just not make a new function that needs to be deprecated the next week.  We will look after deprecating the old ones.